packages/transaction-log/src/components/CustomFilters.tsx

```diff
@@ -0,0 +1,166 @@
+  }
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value)
+  }
+
+  const handleChannelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setChannel(event.target.value)
+  }
+
+  const handleLimitChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setLimit(Number(event.target.value))
+  }
+
+  const handleRerunsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setReruns(event.target.value)
+  }
+
+  return (
+    <Box sx={{padding: '16px'}}>
+      <Grid container spacing={2}>
+        <Grid item xs={2}>
+          <TextField
+            select
+            label="Status"
+            value={status}
+            onChange={handleStatusChange}
+            fullWidth
+          >
+            <MenuItem value="NoFilter">Don't Filter</MenuItem>
+            <MenuItem value="Failed">Failed</MenuItem>
+            <MenuItem value="Processing">Processing</MenuItem>
+            <MenuItem value="Completed">Completed</MenuItem>
+            <MenuItem value="Completed with error(s)">
+              Completed with error(s)
+            </MenuItem>
+            <MenuItem value="Success">Success</MenuItem>
+          </TextField>
+        </Grid>
+        <Grid item xs={2}>
+          <TextField
+            select
+            label="Channel"
+            value={channel}
+            onChange={handleChannelChange}
+            fullWidth
+            InputLabelProps={{shrink: true}}
+          >
+            <MenuItem value="NoFilter">Don't Filter</MenuItem>
+            {channels.map(channel => (
+              <MenuItem key={channel._id} value={channel._id}>
+                {channel.name}
+              </MenuItem>
+            ))}
+          </TextField>
+        </Grid>
+        <Grid item xs={1}>
+          <LocalizationProvider dateAdapter={AdapterDateFns}>
+            <DateTimePicker
+              label="Start Date"
+              value={startDate}
+              onChange={newValue => {
+                setStartDate(newValue)
+              }}
+            />
+          </LocalizationProvider>
+        </Grid>
+        <Grid item xs={1}>
+          <LocalizationProvider dateAdapter={AdapterDateFns}>
+            <DateTimePicker
+              label="End Date"
+              value={endDate}
+              onChange={newValue => {
+                setEndDate(newValue)
+              }}
+            />
+          </LocalizationProvider>
+        </Grid>
+        <Grid item xs={1}>
+          <TextField
+            select
+            label="Limit"
+            value={limit}
+            onChange={handleLimitChange}
+            fullWidth
+          >
+            <MenuItem value={10}>10</MenuItem>
+            <MenuItem value={20}>20</MenuItem>
+            <MenuItem value={50}>50</MenuItem>
+            <MenuItem value={100}>100</MenuItem>
+            <MenuItem value={200}>200</MenuItem>
+            <MenuItem value={500}>500</MenuItem>
```

Comment:
Dont you need key here? example:

`<MenuItem key={10} value={10}>10</MenuItem>`